internal/metrics/metrics.go

```diff
@@ -0,0 +1,128 @@
+package metrics
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/google/pprof/profile"
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/pflag"
+)
+
+// Start - starts gathering metrics based on the type of metrics and writes metrics to string
+// Stop - stops gathering metrics for the type of metrics specified
+type metricType interface {
+	start()
+	stop()
+	getWriter() *bytes.Buffer
+	getIndex() int
+	getMap() map[string]float64
+}
+
+// Metrics - structure to keep information relevant to the metrics calculation
+// Disable - disables metric calculations
+type Metrics struct {
+	metric    metricType
+	metricsID string
+	location  string
+	Disable   bool
+}
+
+// InitializeMetrics - creates a new instance of a Metrics based on the type of metrics specified
+func InitializeMetrics(metric *pflag.Flag) (*Metrics, error) {
```

Comment:
If the change proposed to remove metrics from global is accepted, I think this could return just the error, since MetricsHandler can be initialized inside metrics package...